scripts/bert/embedding.py

```diff
@@ -0,0 +1,168 @@
+        :type sentences: List[str]
+        :param oov: use **sum** or **last** to get token embedding for those out of vocabulary words
+        :type oov: str
+        :return: List[(sentence embedding, tokens, tokens embedding)]
+        """
+        data_iter = self.data_loader(sentences=sentences, batch_size=self.batch_size)
+        batches = []
+        for token_ids, valid_length, token_types in data_iter:
+            token_ids = token_ids.as_in_context(self.ctx)
+            valid_length = valid_length.as_in_context(self.ctx)
+            token_types = token_types.as_in_context(self.ctx)
+            sequence_outputs, pooled_outputs = self.bert(token_ids,
+                                                         token_types,
+                                                         valid_length.astype('float32'))
+            for token_id, sequence_output, pooled_output in zip(token_ids.asnumpy(),
+                                                                sequence_outputs.asnumpy(),
+                                                                pooled_outputs.asnumpy()):
+                batches.append((token_id, sequence_output, pooled_output))
+        if oov == 'sum':
+            return self.oov_sum(batches)
+        else:
+            return self.oov_last(batches)
+
+    def data_loader(self, sentences, batch_size, shuffle=False):
+        tokenizer = BERTTokenizer(self.vocab)
+        transform = BERTSentenceTransform(tokenizer=tokenizer,
+                                          max_seq_length=self.max_seq_length,
+                                          pair=False)
+        dataset = BertEmbeddingDataset(sentences, transform)
+        return DataLoader(dataset=dataset, batch_size=batch_size, shuffle=shuffle)
+
+    def oov_sum(self, batches):
+        """
+        Sum all tensors for oov tokens
+
+        Parameters
+        ----------
+        batches : List[(tokens_id, sequence_outputs, pooled_output].
+            batch
+        """
+
+        # batch:
+        #   token_ids (max_seq_length, ),
+        #   sequence_outputs (max_seq_length, dim, )
+        #   pooled_output (dim, )
+        sentences = []
+        for token_ids, sequence_outputs, pooled_output in batches:
+            tokens = []
+            tensors = []
+            for token_id, sequence_output in zip(token_ids, sequence_outputs):
+                if token_id == 1:
+                    break
+                if token_id in (2, 3):
+                    continue
+                token = self.vocab.idx_to_token[token_id]
+                if token.startswith('##'):
+                    token = token[2:]
+                    tokens[-1] += token
+                    tensors[-1] += sequence_output
+                else:
+                    tokens.append(token)
+                    tensors.append(sequence_output)
+            sentences.append((pooled_output, tokens, tensors))
+        return sentences
+
+    def oov_last(self, batches):
+        """
```

Comment:
The two functions ` oov_sum ` and ` oov_last ` differ only in one line of code.
```python
tensors[-1] += sequence_output
```
or 
```python
tensors[-1] = sequence_output
```
I think we can unify these two functions into one, and then choose to execute different code through a parameter such as ` oov_way `  in the function. 
```python
if oov_way == 'sum':
    tensors[-1] += sequence_output
elif oov_way == 'last':
    tensors[-1] = sequence_output
```
This will make the code more concise.